R/data-repository.R

```diff
@@ -0,0 +1,28 @@
+#' @title DataRepository
+#' @docType class
+#' @description  An object typically holding observed data
+#' @format NULL
+DataRepository <- R6::R6Class(
+  "DataRepository",
+  inherit = DotNetWrapper,
+  cloneable = FALSE,
+  active = list(
+    #' @field baseGrid Returns the base column for the population (typically time column)
+    baseGrid = function(value) {

```

Comment:
in core, we have basegrid, columns, and allButBaseGrid.
I will implement the same methods (baseGrid is typically time, columns are all columns in the dataRepo, allButBaseGrids are all except basegrid)

Should we have a property called Time that returned baseGrid? or property called Measurement that returns col2 and a property called error that returns col3? not sure

Reply:
>Should we have a property called Time that returned baseGrid? or property called Measurement that returns col2 and a property called error that returns col3?

I would say no and no. Basegrid might be different from "Time", we could have several measurements and several errors, so I wouldn't provide shortcuts for the specific use case "_ObservedDataRepo = {Time, Measurement, Error}_"

Reply:
In theory results from a simulation etc.. are DataRepositories
So yes we can have one time and multiple measurements. Each measurements can in theory have multiple errors. I think we are using this for the PI results. 
Practically probably not